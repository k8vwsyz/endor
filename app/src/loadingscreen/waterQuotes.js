export const quotes = [
  { quote: "If there is magic on this planet, it is contained in water.", author: "Loren Eiseley" },
  { quote: "Water is the driving force of all nature.", author: "Leonardo da Vinci" },
  { quote: "Nothing is softer or more flexible than water, yet nothing can resist it.", author: "Lao Tzu" },
  { quote: "We forget that the water cycle and the life cycle are one.", author: "Jacques Yves Cousteau" },
  { quote: "A river seems a magic thing. A magic, moving, living part of the very earth itself.", author: "Laura Gilpin" },
  { quote: "In one drop of water are found all the secrets of all the oceans.", author: "Kahlil Gibran" },
  { quote: "The cure for anything is salt water: sweat, tears or the sea.", author: "Isak Dinesen" },
  { quote: "Thousands have lived without love, not one without water.", author: "W. H. Auden" },
  { quote: "Many a calm river begins as a turbulent waterfall, yet none hurtles and foams all the way to the sea.", author: "Mikhail Lermontov" },
  { quote: "A drop of water, if it could write out its own history, would explain the universe to us.", author: "Lucy Larcom" },
  { quote: "Where the waters do agree, it is quite wonderful the relief they give.", author: "Jane Austen" },
  { quote: "The fall of dropping water wears away the Stone.", author: "Lucretius" },
  { quote: "A lake carries you into recesses of feeling otherwise impenetrable.", author: "William Wordsworth", },
  { quote: "They both listened silently to the water, which to them was not just water, but the voice of life, the voice of Being, the voice of perpetual Becoming.", author: "Hermann Hesse" },
  { quote: "In rivers, the water that you touch is the last of what has passed and the first of that which comes; so with present time.", author: "Leonardo da Vinci" },
  { quote: "When the well’s dry, we know the worth of water.", author: "Benjamin Franklin" },
  { quote: "Water is sufficient…the spirit moves over water.", author: "Friedrich Nietzsche" },
  { quote: "This life is like a swimming pool. You dive into the water, but you can’t see how deep it is.", author: "Dennis Rodman" },
  { quote: "You swam in a river of chance and coincidence. You clung to the happiest accidents—the rest you let float by.", author: "David Wroblewski" },
  { quote: "It is life, I think, to watch the water. A man can learn so many things.", author: "Nicholas Sparks" },
  { quote: "When life places stones in your path, be the water. A persistent drop of water will wear away even the hardest stone.", author: "Autumn Morning Star" },
  { quote: "Human nature is like water. It takes the shape of its container.", author: "Wallace Stevens" },
  { quote: "When you do things from your soul, you feel a river moving in you, a joy.", author: "Rumi" },
  { quote: "You are water I’m water we’re all water in different containers that’s why it’s so asy to meet someday we’ll evaporate together.", author: "Yoko Ono" },
  { quote: "Water is the most critical resource issue of our lifetime and our children’s lifetime. The health of our waters is the principal measure of how we live on the land.", author: "Luna Leopold" },
  { quote: "All water is holy water.", author: "Rajiv Joseph" },
  { quote: "I love the sounds and the power of pounding water, whether it is the waves or a waterfall.", author: "Mike May" },
  { quote: "Until justice rolls down like water and righteousness like a mighty stream.", author: "Martin Luther King Jr" },
  { quote: "Do not feel sad for your tears as rocks never regret the waterfalls", author: "Munia Khan" },
  { quote: "Love, like a river, will cut a new path whenever it meets an obstacle.", author: "Crystal Middlemas" },
  { quote: "Run wild and free like a waterfall", author: "Anamika Mishra" },
  { quote: "Life is like the ocean, it goes up and down.", author: "Vanessa Paradis" },
  { quote: "Always be like a water. Float in the times of pain or dance like waves along the wind which touches its surface.", author: "Santosh Kalwar" },
  { quote: "I don’t drink water because fish fuck in it.", author: "W.C. Fields" },
  { quote: "Don't Eat the Yellow Snow", author: "Frank Zappa" },
];
