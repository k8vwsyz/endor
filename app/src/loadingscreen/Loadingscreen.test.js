import { cleanup, render } from "@testing-library/react";
import React from "react";
import LoadingScreen from "./Loadingscreen";
import * as quotes from "./waterQuotes";


// Change to only 1 quote with 1 author so that
// the snapshot will always match
quotes.quotes = [
    { quote: "quote text", author: "author" }
]

afterEach(cleanup);

describe("Snapshot tests", () => {
    it("check appearance", () => {
        const { asFragment } = render(<LoadingScreen />);
        expect(asFragment()).toMatchSnapshot();
    });
});

describe("Check that stuff appears", () => {
    it("author appears", async () => {
        const { getByTestId } = render(<LoadingScreen />);
        expect(getByTestId("author")).toBeInTheDocument();
    });
    it("quote appears", () => {
        const { getByTestId } = render(<LoadingScreen />);
        expect(getByTestId("quote")).toBeInTheDocument();
    });
    it("loading screen appears", () => {
        const { getByTestId } = render(<LoadingScreen />);
        expect(getByTestId("loadingindicator")).toBeInTheDocument();
    });
});