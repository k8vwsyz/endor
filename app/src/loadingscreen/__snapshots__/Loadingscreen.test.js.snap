// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests check appearance 1`] = `
<DocumentFragment>
  <div>
    <div
      aria-hidden="true"
      class="MuiBackdrop-root makeStyles-backdrop-1"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="makeStyles-root-2"
      >
        <h3
          class="makeStyles-author-3"
          data-testid="author"
        >
          author
        </h3>
        <h3
          class="makeStyles-quote-4"
          data-testid="quote"
        >
          quote text
        </h3>
        <div
          class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
          data-testid="loadingindicator"
          role="progressbar"
        >
          <div
            class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
          />
          <div
            class="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
